### PR TITLE
Add io.github.halfmexican.Mingle

### PR DESCRIPTION
**Pull request by halfmexican**
# Mingle 

A simple GTK4 app to play with Google's Emoji Kitchen.

### Please confirm your submission meets all the criteria

- [X] Please describe your application briefly.
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:https://github.com/halfmexican/mingle**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: com.github.halfmexican.Mingle.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
+    "app-id" : "com.github.halfmexican.Mingle",
```
Comment (bbhtt) [inline com.github.halfmexican.Mingle.json:2]: It should be io.github, the rules are right there in the PR checklist.

File: com.github.halfmexican.Mingle.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        },
+        {
+            "name" : "mingle",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/halfmexican/mingle.git",
+                    "branch": "main"
```
Comment (bbhtt) [inline com.github.halfmexican.Mingle.json:54]: Use a tag and a commit, branch should not be used.

File: com.github.halfmexican.Mingle.metainfo.xml.in
Comment (bbhtt) [inline com.github.halfmexican.Mingle.metainfo.xml.in:1]: This should be upstream

File: com.github.halfmexican.Mingle.metainfo.xml.in [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+	<metadata_license>CC0-1.0</metadata_license>
+	<project_license>GPL-3.0-or-later</project_license>
+	<description>
+	  <p>Combine emojis with Google's Emoji Kitchen, and send them to friends. Written in Vala + GTK4</p>
+	</description>
+  <content_rating type="oars-1.1">
+    <content_attribute id="drugs-alcohol">mild</content_attribute>
+    <content_attribute id="drugs-tobacco">mild</content_attribute>
+    <content_attribute id="sex-themes">mild</content_attribute>
+    <content_attribute id="language-humor">mild</content_attribute>
```
Comment (bbhtt) [inline com.github.halfmexican.Mingle.metainfo.xml.in:16]: Missing screenshots, releases https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/

File: com.github.halfmexican.Mingle.json (hunk 1 above)
Comment (halfmexican) [reply com.github.halfmexican.Mingle.json:2]: I feel so dumb, sorry

File: com.github.halfmexican.Mingle.metainfo.xml.in [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+	<id>com.github.halfmexican.Mingle.desktop</id>
```
Comment (hfiguiere) [inline com.github.halfmexican.Mingle.metainfo.xml.in:3]: id use the reversed dns names NEVER ends in `.desktop`

File: com.github.halfmexican.Mingle.metainfo.xml.in
Comment (hfiguiere) [reply com.github.halfmexican.Mingle.metainfo.xml.in:1]: it is upstream. This one which is `.in` should just be removed.

File: com.github.halfmexican.Mingle.metainfo.xml.in (hunk 3 above)
Comment (hfiguiere) [reply com.github.halfmexican.Mingle.metainfo.xml.in:16]: also missing developer info (`developer_name` or `devloper` tags) and `url`.

File: com.github.halfmexican.Mingle.metainfo.xml.in (hunk 4 above)
Comment (hfiguiere) [reply com.github.halfmexican.Mingle.metainfo.xml.in:3]: this must be fixed upstream when you fix the app-id.

File: com.github.halfmexican.Mingle.json [hunk 5]
```diff
@@ -0,0 +1,59 @@
+{
+    "app-id" : "com.github.halfmexican.Mingle",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "46",
+    "sdk" : "org.gnome.Sdk",
+    "sdk-extensions" : [
+        "org.freedesktop.Sdk.Extension.vala"
```
Comment (hfiguiere) [inline com.github.halfmexican.Mingle.json:7]: I don't think this is necessary. The GNOME sdk contains Vala.

File: com.github.halfmexican.Mingle.json (hunk 1 above)
Comment (halfmexican) [reply com.github.halfmexican.Mingle.json]: changed the entire App ID

File: com.github.halfmexican.Mingle.metainfo.xml.in (hunk 3 above)
Comment (halfmexican) [reply com.github.halfmexican.Mingle.metainfo.xml.in]: I've removed this file from the pr but added all the requested info and ran `appstreamcli validate`

File: com.github.halfmexican.Mingle.metainfo.xml.in (hunk 4 above)
Comment (halfmexican) [reply com.github.halfmexican.Mingle.metainfo.xml.in]: fixed!

File: com.github.halfmexican.Mingle.json (hunk 5 above)
Comment (halfmexican) [reply com.github.halfmexican.Mingle.json]: removed

**Comment by halfmexican** [comment]
1. I have [changed the app-id](https://github.com/halfmexican/mingle/pull/14) everywhere were needed upstream and tested that the application still built. 
2. added [releases](https://github.com/halfmexican/mingle/commit/fdc45b0e977e10f110d65944b2dda3db60603930), [screenshots](https://github.com/halfmexican/mingle/commit/fdaacaecea3131725643680250006ed58d3d7c1f), developer, and branding colors to metainfo and removed the file from this pr
3. removed the vala extension and used a commit and tag instead of branch

**Comment by hfiguiere** [comment]
bot, build io.github.halfmexican.Mingle

File: io.github.halfmexican.Mingle.json [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland"
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/vala/bin",
+        "prepend-ld-library-path" : "/usr/lib/sdk/vala/lib"
+    },
```
Comment (hfiguiere) [inline io.github.halfmexican.Mingle.json:17]:
```suggestion
```

since you removed the SDK.

File: io.github.halfmexican.Mingle.json [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        {
+            "name" : "mingle",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/halfmexican/mingle.git",
+                    "commit": "fdaacaecea3131725643680250006ed58d3d7c1f",
+                    "tag":  "v0.10"
```
Comment (hfiguiere) [inline io.github.halfmexican.Mingle.json:48]: this tag doesn't exist.

Comment (halfmexican) [reply io.github.halfmexican.Mingle.json:48]: I'm so sorry for my incompetence.
it should exist [now](https://github.com/halfmexican/mingle/releases/tag/v0.10)

**Comment by bbhtt** [comment]
bot, build io.github.halfmexican.Mingle

**Pull request closed without merging**